modules/rostests/apitests/shell32/ShellState.cpp

```diff
@@ -0,0 +1,226 @@
+/*
+ * PROJECT:         ReactOS API tests
+ * LICENSE:         LGPLv2.1+ - See COPYING.LIB in the top level directory
+ * PURPOSE:         Test for SHELLSTATE
+ * PROGRAMMERS:     Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
+ */
+#include "shelltest.h"
+
+#define NDEBUG
+#include <debug.h>
+#include <shellutils.h>
+#include <stdio.h>
+
+/* [HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer] */
+/* The contents of RegValue ShellState. */
+typedef struct REGSHELLSTATE
+{
+    DWORD dwSize;
+    SHELLSTATE ss;
+} REGSHELLSTATE, *PREGSHELLSTATE;
+
+static void dump(const char *name, const void *ptr, size_t siz)
+{
+    const BYTE *pb;
+    printf("%s: ", name);
+
+    pb = (const BYTE *)ptr;
+    while (siz--)
+    {
+        printf("%02X ", *pb++);
+    }
+    printf("\n");
+}
+
+static int read_key(REGSHELLSTATE *prss)
+{
+    HKEY hKey;
+    LONG result;
+    DWORD cb;
+    static const LPCWSTR s_pszExplorer =
+        L"Software\\Microsoft\\Windows\\CurrentVersion\\Explorer";
+
+    memset(prss, 0, sizeof(*prss));
+
+    result = RegOpenKeyExW(HKEY_CURRENT_USER, s_pszExplorer, 0, KEY_READ, &hKey);
+    ok(result == ERROR_SUCCESS, "\n");
```

Comment:
If would be nicer to print a message on failure (ideally containing the actual value of result).

Reply:
OK.